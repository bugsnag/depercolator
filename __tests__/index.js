import fs from 'async-file';
import resolveBin from 'resolve-bin';
import { execAsync } from 'async-child-process';
import path from 'path';

const COMMAND = path.resolve('./dist/index.js');

function mapExtension(source) {
  const ext = path.extname(source);
  return `${path.dirname(source)}/${path.basename(source, ext)}.jsx`;
}

async function run(filename, flags = '') {
  const source = path.resolve(filename);
  const outputPath = mapExtension(source);
  await execAsync(`${COMMAND} ${flags} ${source}`);

  const result = await fs.readFile(outputPath, 'utf8');
  fs.unlink(outputPath);
  return result;
}

describe('happy test', () => {
  it('converts the file', async () => {
    const result = await run('./fixtures/testfile.cjsx');
    expect(result).toMatchSnapshot();
  });
});

describe('invalid option', () => {
  it('throws the error', async () => {
    expect.assertions(1);
    await expect(run('./fixtures/badfile.cjsx')).rejects.toBeDefined();
  });
});

describe('decaffeinate options', () => {
  it('supports all of them', async () => {
    const decafPath = resolveBin.sync('decaffeinate');
    const { stdout } = await execAsync(`${decafPath} -h`);
    const flags = stdout.match(/\s-[\w-]+/g);
    expect(flags).toMatchSnapshot();
  });
});

describe('prettier options', () => {
  it('supports all of them', async () => {
    const prettierPath = resolveBin.sync('prettier');
    const { stdout } = await execAsync(`${prettierPath} -h`);
    const flags = stdout.match(/\s-[\w-]+/g);
    expect(flags).toMatchSnapshot();
  });
});

describe('flags with <int>', () => {
  it('cleans the int', async () => {
    const result = await run('./fixtures/testfile.cjsx', '--tab-width 4');
    expect(result).toMatchSnapshot();
  });
});

describe('flag --use-js-modules', () => {
  it('does not camelcase js', async () => {
    const result = await run('./fixtures/testfile.cjsx', '--use-js-modules');
    expect(result).toMatchSnapshot();
  });
});

describe('flag --loose-js-modules', () => {
  it('does not camelcase js', async () => {
    const result = await run('./fixtures/named_export.cjsx', '--loose-js-modules --use-js-modules');
    expect(result).toMatchSnapshot();
  });
});
