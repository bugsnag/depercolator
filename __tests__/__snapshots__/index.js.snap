// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decaffeinate options supports all of them 1`] = `
Array [
  " -h",
  " --help",
  " --modernize-js",
  " --literate",
  " --disable-suggestion-comment",
  " --no-array-includes",
  " --use-optional-chaining",
  " --use-js-modules",
  " --loose-js-modules",
  " --safe-import-function-identifiers",
  " --prefer-let",
  " --loose",
  " --loose",
  " --loose-default-params",
  " --loose-for-expressions",
  " --loose-for-of",
  " --loose-includes",
  " --loose-comparison-negation",
  " --disable-babel-constructor-workaround",
  " --disallow-invalid-constructors",
]
`;

exports[`flag --loose-js-modules does not camelcase js 1`] = `
"/*
 * decaffeinate suggestions:
 * DS102: Remove unnecessary code created because of implicit returns
 * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
 */
import React from 'react';

const testConstant = 'foo';

const klass = React.createClass({
  render() {
    return (
      <div className=\\"hello\\">
        <div>Hello</div>
      </div>
    );
  },
});

export { klass as named };
"
`;

exports[`flag --use-js-modules does not camelcase js 1`] = `
"/*
 * decaffeinate suggestions:
 * DS102: Remove unnecessary code created because of implicit returns
 * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
 */
import React from 'react';

const testConstant = 'foo';

export default React.createClass({
  render() {
    return (
      <div className=\\"hello\\">
        <div>Hello</div>
      </div>
    );
  },
});
"
`;

exports[`flags with <int> cleans the int 1`] = `
"/*
 * decaffeinate suggestions:
 * DS102: Remove unnecessary code created because of implicit returns
 * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
 */
const React = require('react');

const testConstant = 'foo';

module.exports = React.createClass({
    render() {
        return (
            <div className=\\"hello\\">
                <div>Hello</div>
            </div>
        );
    },
});
"
`;

exports[`happy test converts the file 1`] = `
"/*
 * decaffeinate suggestions:
 * DS102: Remove unnecessary code created because of implicit returns
 * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
 */
const React = require('react');

const testConstant = 'foo';

module.exports = React.createClass({
  render() {
    return (
      <div className=\\"hello\\">
        <div>Hello</div>
      </div>
    );
  },
});
"
`;

exports[`prettier options supports all of them 1`] = `
Array [
  " -l",
  " --list-different",
  " --write",
  " --no-bracket-spacing",
  " --jsx-bracket-same-line",
  " --parser",
  " --print-width",
  " --no-prose-wrap",
  " --no-semi",
  " --single-quote",
  " --tab-width",
  " --trailing-comma",
  " --use-tabs",
  " --config",
  " --no-config",
  " --config-precedence",
  " --find-config-path",
  " --ignore-path",
  " --with-node-modules",
  " --cursor-offset",
  " --range-start",
  " --range-end",
  " -1",
  " --range-end",
  " --cursor-offset",
  " --range-start",
  " --cursor-offset",
  " --no-color",
  " -h",
  " --help",
  " --help",
  " --insert-pragma",
  " --loglevel",
  " --require-pragma",
  " --stdin",
  " --stdin-filepath",
  " --support-info",
  " -v",
  " --version",
]
`;
